specification/resourcegraph/resource-manager/Microsoft.ResourceGraph/preview/2018-09-01-preview/resourcegraph.json

```diff
@@ -88,6 +88,90 @@
         }
       }
     },
+    "/providers/Microsoft.ResourceGraph/resourceChanges": {
+      "post": {
+        "description": "List changes to a resource for a given time interval.",
+        "operationId": "ResourceChanges",
```

Comment:
Looking for guidance. I'm getting the following linter issue here:

> Since operation 'ResourceChanges' response has model definition 'array', it should be of the form "*_list*". Note: If you have already shipped an SDK on top of this spec, fixing this warning may introduce a breaking change.

**Question 1**. Is this rule applicable? [R1003 ListInOperationName](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#R1003) says:

> Per ARM SDK guidelines, each 'GET' operation on a resource should have "list" in the name...

However, this is POST, not GET. Should we just suppress it?

**Question 2**. If the rule is applicable anyway, how should we fix it? Renaming it to **ResourceChanges_List** causes:

> OperationId should contain the verb: 'resourcechanges' in:'ResourceChanges_List'. Consider updating the operationId

Renaming it to **ResourceChanges_ListResourceChanges** causes:

> Per the Noun_Verb convention for Operation Ids, the noun 'ResourceChanges' should not appear after the underscore. Note: If you have already shipped an SDK on top of this spec, fixing this warning may introduce a breaking change.

Renaming it to **ResourceChanges_Listresourcechanges** seems to get rid of warnings, but the result looks weird.

Any guidance?

Reply:
Given that all the other operations in this spec don't follow the `noun_verb` convention I think we shouldn't introduce it here as that would bifurcate the codegen (i.e. the new API would be in its own operation group).